build_badtrack.py

```diff
@@ -0,0 +1,88 @@
+
+import os
+import shutil
+import subprocess
+
+# Get the current working directory
+APP_PATH = os.path.abspath(os.path.dirname(__file__))
+
+# Fixed path for the environment variable folders
+HISTORY_FOLDER = "/var/lib/badtrack/history"
+CACHE_FOLDER = "/var/lib/badtrack/cache"
+
+# Create the badtrack directory structure
+os.makedirs(f"{APP_PATH}/badtrack/DEBIAN", exist_ok=True)
+os.makedirs(f"{APP_PATH}/badtrack/usr/local/bin/badtrack", exist_ok=True)
+os.makedirs(f"{APP_PATH}/badtrack/etc/systemd/system", exist_ok=True)
+os.makedirs(f"{APP_PATH}/badtrack/var/lib/badtrack/history", exist_ok=True)
+os.makedirs(f"{APP_PATH}/badtrack/var/lib/badtrack/cache", exist_ok=True)
+
+# Copy the main.py file to the badtrack directory and make it executable
+shutil.copy('main.py', f"{APP_PATH}/badtrack/usr/local/bin/badtrack/main.py")
+os.chmod(f"{APP_PATH}/badtrack/usr/local/bin/badtrack/main.py", 0o755)
+
+# Set perissions for environment variable folders. Subprocess.run allows recursive chmod where os.chmod doesn't seem to.
+subprocess.run(["chmod", "--recursive", "755", f"{APP_PATH}/badtrack/var/lib/badtrack/history"])
+subprocess.run(["chmod", "--recursive", "755", f"{APP_PATH}/badtrack/var/lib/badtrack/cache"])
+
+# Create the control fi le
+control_content = """\
+Package: badtrack
+Version: 1.0.0
+Section: custom
+Priority: optional
+Architecture: all
+Essential: no
+Installed-Size: 1024
+Maintainer: Your Name <your-email@example.com>
+Description: Badtrack is a sample package
+"""
+with open(f"{APP_PATH}/badtrack/DEBIAN/control", 'w') as file:
+    file.write(control_content)
+
+# Create the systemd service file within the package
+service_content = f"""\
+[Unit]
+Description=BadTrack Service
+After=network.target
+
+[Service]
+Type=simple
+User=badtrackuser
+WorkingDirectory=/usr/local/bin/badtrack
+ExecStart=/usr/bin/python3 /usr/local/bin/badtrack/main.py
+Environment=HISTORY_FOLDER={HISTORY_FOLDER}
+Environment=CACHE_FOLDER={CACHE_FOLDER}
+
+[Install]
+WantedBy=multi-user.target
+"""
+with open(f"{APP_PATH}/badtrack/etc/systemd/system/badtrack.service", 'w') as file:
+    file.write(service_content)
+
+# Changing the permissions of the badtrack.service file
+os.chmod(f"{APP_PATH}/badtrack/etc/systemd/system/badtrack.service", 0o644)
+
+# Create the post-installation script
+postinst_content = f"""\
+#!/bin/bash
+getent passwd badtrackuser > /dev/null || sudo useradd -r -s /bin/false badtrackuser
+# Set ownership of folders to badtrackuser
+chown -R badtrackuser:badtrackuser \"{HISTORY_FOLDER}\"
+chown -R badtrackuser:badtrackuser \"{CACHE_FOLDER}\"
+systemctl enable badtrack
+systemctl restart badtrack
+systemctl daemon-reload
+"""
+
+with open(f"{APP_PATH}/badtrack/DEBIAN/postinst", 'w') as file:
+    file.write(postinst_content)
+
+# Make the post-installation script executable
+os.chmod(f"{APP_PATH}/badtrack/DEBIAN/postinst", 0o755)
+
+# Build the Debian package
+subprocess.run(["dpkg-deb", "--build", f"{APP_PATH}/badtrack"])
```

Comment:
`subprocess.run(["dpkg-deb", "--build", f"{APP_PATH}/badtrack"], check=True)` The check causes Python exception if command fails.